packages/xstate-store/src/store.ts

```diff
@@ -95,7 +116,23 @@ function createStoreCore<
     observers?.forEach((o) => o.next?.(currentSnapshot));
   }
 
-  const store: Store<TContext, StoreEvent> = {
+  const store: Store<TContext, StoreEvent, TEmitted> = {
+    on: (event, fn) => {
+      if (!listeners) {
+        listeners = new Map();
+      }
+      if (!listeners.has(event)) {
+        listeners.set(event, new Set());
+      }
+      const set = listeners.get(event)!;
+      set.add(fn);
+
+      return {
+        unsubscribe() {
+          set.delete(fn);
+        }
+      };
+    },
```

Comment:
please follow the binding scheme used by `createActor` here: https://github.com/statelyai/xstate/blob/5afff2cd0662347b6c031a3362392eb33ad15da0/packages/core/src/createActor.ts#L423-L442

this has some semantic differences and that binding scheme improves compat with the observable spec proposal